src/controller.cpp

```diff
@@ -51,9 +51,9 @@ geometry_msgs::msg::TwistStamped Controller::computeVelocityCommands(
   const geometry_msgs::msg::PoseStamped & robot_pose, const geometry_msgs::msg::Twist & robot_speed,
   nav2_core::GoalChecker * /*goal_checker*/)
 {
-  geometry_msgs::msg::TwistStamped cmd;
-  nav_msgs::msg::Path transformed_plan = path_handler_.transformPath(robot_pose);
-  cmd = optimizer_.evalControl(robot_pose, robot_speed, transformed_plan);
+  nav_msgs::msg::Path && transformed_plan = path_handler_.transformPath(robot_pose);
```

Comment:
But there is no auto in this statement, the output type of `transformPath` and the type of `transformed_plan` are both explicitly typed.

Reply:
Sure. but when you use auto var = getSomething() compiler tries to deduce type of var, and it could deduce this to reference or to value, copy right hand side, or move it. 
This analogy with templates shows the deduced types of T and deduced type of auto var in cases when you pass lvalue (variable that has name) and rvalue (variable that has no name, e.g output of the function, or std::move things)